tests/storage/test_mount_sanity.py

```diff
@@ -0,0 +1,79 @@
+import threading
+import subprocess
+import time
+import sys
+sys.path.append("..")
+import _thread as thread
+import os
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+from test_main import Test
+import argparse
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--timeout", dest="timeout", type=int,
+                        required=True,
+                        help = 'Timeout for the function')
+    parser.add_argument("--mount_points", nargs='+', dest="mount_points", type=str,
+                        required=True,
+                        help='Mount Points')
+    args = parser.parse_args()
+    return args
+
+class MountSanity(Test):
+    """
+    Checks Sanity of mount points
+
+    Command - python3 test_mount_sanity.py --timeout 5 --mount_points user user/u
+
+    """
+    def __init__(self, timeout, mount_points):
+        self.mount_points = mount_points
+        self.timeout = timeout
+        params = {}
+        params['test_name'] = "mount_sanity"
+        Test.__init__(self, **params)
+
+
+    def quit_function(self, fn_name):
+        self.log.write("error", '{0} took too long'.format(fn_name))
+        self.exit = 1
+        sys.stderr.flush()  # Python 3 stderr is likely buffered.
+        thread.interrupt_main()  # raises KeyboardInterrupt if function is taking too long
+
+    def check_mount(self, timeout):
```

Comment:
Can you explain me a bit the flow of this method?
I read it quickly but I got the impression that you kill the entire framework if one of the mounts is stuck. Is this the case? Should it be?

Reply:
Yes, currently this is the case. As it was discussed in the meeting that, in the past some mountpoints were badly hanging and all the processes trying to access them became zombie processes.

Reply:
Yes, but should we kill the entire framework or just the current test?

Reply:
I might be wrong here, but considering EOS provides storage to both CERNBox and SWAN, I feel it's better to kill the framework because our whole testing framework might go down. As an example if we go about killing the current test (i.e storage/mount), and we continue running the other tests for instance, tests for JupyterhubAPI and then we try to spawn a session (lot of things happen under the hood like some storage would be allocated for the particular user), then what if spawn process becomes a zombie process because it tried to access one of those mount points?

Reply:
Yes, but we have different mountpoints. If one is broken, we don't know about the others. And we also might test CVMFS. And if EOS is down, maybe it would be interesting to see how JH or the user containers react. I'm not sure on what to do, though.

Reply:
Opened an issue about this